### PR TITLE
executor: add more diagnosis rule to check some metrics exceed thresholds

### PR DESCRIPTION
**Pull request by crazycs520**
Signed-off-by: crazycs <crazycs520@gmail.com>

### What problem does this PR solve? 
 This PR is the continuation of  PR https://github.com/pingcap/tidb/pull/14801.

This PR adds a more `threshold-check` rule, which is used to check some metrics threshold value, like `tso duration`, `get token duration`, `storage-write` duration and so on. It will detect the following metrics tables in the current implementation:

* `pd_scheduler_store_status`
* `pd_region_health`

### What is changed and how it works?

Get the metric-data and compare with the threshold.

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: executor/inspection_result.go [hunk 1]
```diff
@@ -663,3 +665,139 @@ func (thresholdCheckInspection) inspectThreshold2(ctx context.Context, sctx sess
...
+				(t1.value-t2.value)/t1.value>%v;`, condition, c.tp, c.threshold)
+}
+
+func (c compareStoreStatus) genResult(_ string, row chunk.Row) inspectionResult {
+	addr1 := row.GetString(0)
+	value1 := row.GetFloat64(1)
+	addr2 := row.GetString(2)
+	value2 := row.GetFloat64(3)
+	ratio := row.GetFloat64(4)
+	expected := fmt.Sprintf("The difference between %v and %v should less than %v%%, actual is %v%%", addr1, addr2, c.threshold*100, ratio*100)
```
Comment (lonng) [inline executor/inspection_result.go]: I think the `expected` is too verbose.
We can rename the `item` to `region-score-balance` and set the `expected` to the threshold,  the `ratio` information should be put in `detail`.

File: executor/inspection_result_test.go [hunk 2]
```diff
@@ -368,12 +373,74 @@ func (s *inspectionResultSuite) TestThresholdCheckInspection2(c *C) {
...
+		"tikv_handle_snapshot_duration":       {},
+		"tikv_storage_async_request_duration": {},
+		"tikv_engine_write_duration":          {},
+		"tikv_engine_max_get_duration":        {},
+		"tikv_engine_max_seek_duration":       {},
+		"tikv_scheduler_pending_commands":     {},
+		"tikv_block_index_cache_hit":          {},
+		"tikv_block_data_cache_hit":           {},
+		"tikv_block_filter_cache_hit":         {},
+		"pd_scheduler_store_status": {
```
Comment (lonng) [inline executor/inspection_result_test.go]: You don't mock this data and just rewrite your SQL with an addtional condition `where item in ('leader_score', '...')`

Comment (crazycs520) [reply executor/inspection_result_test.go]: great, done.

**Pull request merged**